tests/testsuite/publish.rs

```diff
@@ -1686,6 +1686,410 @@ repository = "foo"
     );
 }
 
+#[cargo_test]
+fn publish_with_features() {
+    let registry = RegistryBuilder::new().http_api().http_index().build();
+
+    Package::new("common", "1.0.0").feature("f1", &[]).publish();
+    Package::new("baz", "1.0.0").feature("cat", &[]).publish();
+    Package::new("bat", "1.0.0").feature("cat", &[]).publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [package]
+            name = "foo"
+            version = "0.1.0"
+            edition = "2015"
+            authors = []
+            license = "MIT"
+            description = "foo"
+            documentation = "foo"
+            homepage = "foo"
+            repository = "foo"
+
+
+            [features]
+            foo_feature = ["bar/cat", "common/f1", "baz/cat", "bat/cat"]
+
+            [dependencies]
+            common = { version = "1.0", features = ["f1"] }
+            bat = { version = "1.0", features = ["cat"] }
+
+            [build-dependencies]
+            baz = { version = "1.0", features = ["cat"] }
+
+
+            [dev-dependencies.bat]
+            version = "1.0"
+            features = ["cat"]
+
+            [dev-dependencies.bar]
+            path = "../bar"
+            features = ["cat"]
```

Comment:
Could you name the dependencies relative to what role they fulfill in the test to make the intent more clear at a glance?

Reply:
Updated more clear name.

Reply:
Could you clean up the commits?

Reply:
Sure, updated.